types/src/verifier/mod.rs

```diff
@@ -176,10 +177,9 @@ pub trait ProofVerifier {
         let prover_id = prover_id_from_u64(prover);
 
         // Verify Proof
-        if !post::verify_window_post(&bytes_32(&randomness), &proofs, &replicas, prover_id)? {
-            Err("Window post was invalid".into())
-        } else {
-            Ok(())
+        match post::verify_window_post(&bytes_32(&randomness), &proofs, &replicas, prover_id) {
```

Comment:
If you only want to transform `Err` case and leave the `Ok` case untouched you can use:
```
post::verify_window_post(&bytes_32(&randomness), &proofs, &replicas, prover_id)
    .map_err(|err| err.into())
```